mathesar_ui/src/sections/table-view/header/header-cell/type-options/TypeOptions.svelte

```diff
@@ -0,0 +1,151 @@
+
+  $: allowedTypeConversions = ColumnsDataStore.getAllowedTypeConversions(
+    column,
+    $abstractTypes.data,
+  );
+
+  let selectedAbstractType: AbstractType = null;
+  let selectedDBTypeOption: SelectOption<DbType> = null;
+  let typeChangeState = States.Idle;
+
+  function selectAbstractType(abstractType: AbstractType) {
+    selectedAbstractType = abstractType;
+    if (abstractType.identifier === abstractTypeOfColumn.identifier) {
+      selectedDBTypeOption = {
+        id: column.type,
+        label: column.type,
+      };
+    } else if (abstractType.defaultDbType) {
+      selectedDBTypeOption = {
+        id: abstractType.defaultDbType,
+        label: abstractType.defaultDbType,
+      };
+    }
+  }
+
+  function resetAbstractType() {
+    selectedAbstractType = abstractTypeOfColumn;
+    selectedDBTypeOption = {
+      id: column.type,
+      label: column.type,
+    };
+  }
+
+  async function scrollToSelectedType() {
+    await tick();
+    const selectedElement: HTMLLIElement = abstractTypeContainer?.querySelector('li.selected');
+    if (selectedElement) {
+      abstractTypeContainer.scrollTop = selectedElement.offsetTop;
+    }
+  }
+
+  $: if (!selectedAbstractType && abstractTypeOfColumn) {
+    resetAbstractType();
+    void scrollToSelectedType();
+  }
+
+  function calculateDBTypeOptions(_selectedAbstractType: AbstractType): SelectOption[] {
+    if (_selectedAbstractType) {
+      return Array.from(_selectedAbstractType?.dbTypes).map((entry) => ({
+        id: entry,
+        label: entry,
+      }));
+    }
+    return [];
+  }
+
+  $: dbTypeOptions = calculateDBTypeOptions(selectedAbstractType);
+
+  function close() {
+    resetAbstractType();
+    typeChangeState = States.Done;
+    dispatch('close');
+  }
+
+  async function onSave() {
+    if (selectedDBTypeOption.id !== column.type) {
+      typeChangeState = States.Loading;
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      await columnsDataStore.patchType(column.index, selectedDBTypeOption.id);
+    }
+    close();
+  }
+</script>
+
+<div class="column-type-menu">
+  <h5 class="menu-header">Set Column Type</h5>
+  <ul bind:this={abstractTypeContainer} class="type-list">
+    {#each allowedTypeConversions as abstractType (abstractType.identifier)}
+      <li class:selected={selectedAbstractType?.identifier === abstractType?.identifier}>
+        <Button appearance="plain" on:click={() => selectAbstractType(abstractType)}>
+          <span class="data-icon">{abstractType.icon}</span>
+          <span>{abstractType.name}</span>
+        </Button>
+      </li>
+    {/each}
+  </ul>
+  
+  <div class="type-options">
+    <!-- TODO: Make tab container more generic to be used here -->
+    <ul class="type-option-tabs">
+      <li>
+        <Button appearance="ghost" class="padding-zero type-option-tab">
+          <Icon size="0.75em" data={faDatabase}/>
+          <span>Database</span>
+        </Button>
```

Comment:
Using a `Button` seems strange here because it gives a `pointer` cursor but has no function when clicked.

Reply:
There is a comment above that block which says: `TODO: Make tab container more generic to be used here 
`

As per [our UX](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=4270%3A39549&starting-point-node-id=4270%3A39549), this should be a tab with a couple options. However, our current Tab component is not generic enough to handle this. I've used a button for the time being.

As we discussed in one of our previous conversations, we will have to create a generic Tab component in the library and a mathesar specific one which builds over it. We should be able to use the generic Tab component here.